app/javascript/pages/admin/Tags/edit.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { graphql, compose } from 'react-apollo'
+import { NetworkStatus } from 'apollo-client'
+import { connect } from 'react-redux'
+import * as R from 'ramda'
+
+import { graphQLError } from 'actions'
+
+import TagForm from './form'
+import Loading from 'components/LoadingIcon'
+
+import TagQuery from './queries/show.gql'
+import UpdateTagMutation from './mutations/update.gql'
+
+const EditTag = ({ data: { networkStatus, tag }, updateTag }) =>
+  networkStatus === NetworkStatus.loading ? <Loading /> : <TagForm tag={tag} onSubmit={updateTag} />
+
+const buildOptimisticResponse = tag => ({
+  __typename: 'Mutation',
+  updateTag: {
+    __typename: 'tag',
+    ...tag,
+  },
+})
+
+const withData = compose(
```

Comment:
can we use `apollo-hooks` instead of `react-apollo` please? I can pair with you if you like.

Reply:
sure, let's pair on it